modules/xfem/src/base/XFEM.C

```diff
@@ -967,6 +989,31 @@ XFEM::cutMeshWithEFA()
 
       libmesh_elem->set_node(j) = libmesh_node;
 
+      // Store solution for all nodes affected by XFEM (even existing nodes)
+      if (parent_elem->is_semilocal(_mesh->processor_id()))
```

Comment:
The semilocal element might be attached to local nodes.  We err on the side of storing more data than we actually need here, and then only do the setting of the solution on entities that are locally owned.  This is necessary for this to work in cases where the newly-created nodes end up on different processors.